packages/discovery-provider/plugins/notifications/src/tasks/dmNotifications.ts

```diff
@@ -115,6 +116,106 @@ async function getUnreadReactions(
     .andWhereRaw('chat_message_reactions.user_id != chat_member.user_id')
 }
 
+async function getNewBlasts(
+  discoveryDB: Knex,
+  minTimestamp: Date,
+  maxTimestamp: Date
+): Promise<DMNotification[]> {
+  const messages = await discoveryDB.raw(
+    `
+    WITH blast AS (
+      SELECT * FROM chat_blast
+    ),
+    aud as (
+      -- follower_audience
+      SELECT blast_id, follower_user_id AS to_user_id
+      FROM follows
+      JOIN blast
+        ON blast.audience = 'follower_audience'
+        AND follows.followee_user_id = blast.from_user_id
+        AND follows.is_delete = false
+        AND follows.created_at < blast.created_at
+
+      UNION
+
+      -- tipper_audience
+      SELECT blast_id, sender_user_id AS to_user_id
+      FROM user_tips tip
+      JOIN blast
+        ON blast.audience = 'tipper_audience'
+        AND receiver_user_id = blast.from_user_id
+        AND tip.created_at < blast.created_at
+
+      UNION
+
+      -- remixer_audience
+      SELECT blast_id, t.owner_id AS to_user_id
+      FROM tracks t
+      JOIN remixes ON remixes.child_track_id = t.track_id
+      JOIN tracks og ON remixes.parent_track_id = og.track_id
+      JOIN blast
+        ON blast.audience = 'remixer_audience'
+        AND og.owner_id = blast.from_user_id
+        AND (
+          blast.audience_content_id IS NULL
+          OR (
+            blast.audience_content_type = 'track'
+            AND blast.audience_content_id = og.track_id
+          )
+        )
+
+      UNION
+
+      -- customer_audience
+      SELECT blast_id, buyer_user_id AS to_user_id
+      FROM usdc_purchases p
+      JOIN blast
+        ON blast.audience = 'customer_audience'
+        AND p.seller_user_id = blast.from_user_id
+        AND (
+          blast.audience_content_id IS NULL
+          OR (
+            blast.audience_content_type = p.content_type::text
+            AND blast.audience_content_id = p.content_id
+          )
+        )
+    ),
+      targ AS (
+      SELECT
+        blast_id,
+        from_user_id,
+        to_user_id,
+        blast.created_at
+      FROM blast
+      JOIN aud USING (blast_id)
+      LEFT JOIN chat_member member_a on from_user_id = member_a.user_id
+      LEFT JOIN chat_member member_b on to_user_id = member_b.user_id and member_b.chat_id = member_a.chat_id
+      WHERE date_trunc('milliseconds', blast.created_at) > greatest(member_b.last_active_at, ?)
+      AND date_trunc('milliseconds', blast.created_at) <= ?
+    )
+    SELECT from_user_id as sender_user_id, to_user_id as receiver_user_id, created_at FROM targ;
+    `,
+    [minTimestamp.toISOString(), maxTimestamp.toISOString()]
+  )
+  const formattedMessages = messages.rows.map((message) => {
+    let chatId: string
+    if (message.sender_user_id < message.receiver_user_id) {
```

Comment:
Just double checking this is the ordering of the chat ids? I'm probably misremembering them being sorted by the hashed ids

Reply:
ugh can't share code cuz diff languages